Sources/PT.PM.Matching.Tests/PatternMatchingTests.cs

```diff
@@ -110,6 +110,25 @@ public void Match_PasswordCheckInsideStatement(string patternData)
             Assert.AreEqual(expectedMatchingCount, matchResults.Count());
         }
 
+        [Test]
+        public void Match_PatternAnyWithRegex()
+        {
+            PatternRoot pattern = new PatternRoot
+            {
+                DebugInfo = "Test PatternAny",
+                Languages = new HashSet<Language> { Php.Language },
+                Node = new PatternAny("password")
+            };
+            patternsRepository.Add(patternsConverter.ConvertBack(new List<PatternRoot>() { pattern }));
+
+            WorkflowResult workflowResult = workflow.Process();
+            IEnumerable<MatchResultDto> matchResults = workflowResult.MatchResults.ToDto();
+            patternsRepository.Clear();
+
+            int expectedCount = 3;
```

Comment:
Not correct for such pattern. I see 6 matches in this code:

```php
<?php 
test_call_0();
test_call_1(a);
test_call_2(a, e);
test_call_3(c, d, a);
test_call_4(c, d, a, e);

$password = a;
call_with_password_param(1, $password, 2);

$password2 = "1234";
if ($password2->Length > 0) { }
test_call_5(1, $password2, 2);
$concat = "111" .
   "222222222"   .  
  "333";
```

Also, it's not necessary to declare a variable here.

Reply:
logic changed due to your review